go-controller/pkg/ovn/pods.go

```diff
@@ -280,13 +286,20 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 	}
 
 	// If the pod has not already been assigned addresses, read them now
-	if podMac == nil || podCIDR == nil {
-		var podIP net.IP
-		podMac, podIP, err = waitForPodAddresses(portName)
+	if podMac == nil || podCIDRs == nil {
+		var podIPs []net.IP
+		podMac, podIPs, err = waitForPodAddresses(portName)
 		if err != nil {
 			return err
 		}
-		podCIDR = &net.IPNet{IP: podIP, Mask: nodeSubnet.Mask}
+		for _, podIP := range podIPs {
+			// DUAL-STACK FIXME: need multiple nodeSubnet.Mask
+			mask := nodeSubnet.Mask
```

Comment:
#1283 merged yesterday and now `oc.waitForNodeLogicalSwitch` (which is where `nodeSubnet` came from) is nearly-dual-stack; you just need to fix it to actually return both subnets.

You'll need to match up the `nodeSubnets` with the `podIPs` correctly in terms of IPv4/IPv6; they're not guaranteed to be in the same order. A helper function like `gatewayForSubnet` in `pkg/ovn/gateway_init.go` might be useful.

Reply:
Was next on my list. I'll get right this.